### ISSUE TITLE
Cannot enable OAuth plugin - return 400 Bad request

### ISSUE THREAD

**Issue by milaw** [Kong/kong#4251]
Please read the CONTRIBUTING.md guidelines to learn on which channels you can
seek for help and ask general questions:

https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#where-to-seek-for-help

### Summary

Get 400 Bad Request enabling OAuth plugin on API
Used  2 clients and got the same result

**HTTPie**

http post localhost:8001/apis/test1/plugins name=oauth2 config.enable_client_credentials=true

HTTP/1.1 400 Bad Request
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 485
Content-Type: application/json; charset=utf-8
Date: Mon, 28 Jan 2019 14:50:19 GMT
Server: kong/0.15.0

{
    "code": 2,
    "fields": {
        "config": "at least one of these fields must be true: enable_authorization_code, enable_implicit_grant, enable_client_credentials, enable_password_grant",
        "config.enable_client_credentials": "unknown field"
    },
    "message": "2 schema violations (config: at least one of these fields must be true: enable_authorization_code, enable_implicit_grant, enable_client_credentials, enable_password_grant; config.enable_client_credentials: unknown field)",
    "name": "schema violation"
}

**CURL** 

curl -X POST http://localhost:8001/apis/test1/plugins -d "name=oauth2" --data "config.enable_client_credentials=true"
{"message":"schema violation (config.enable_client_credentials: expected a boolean)","name":"schema violation","fields":{"config":{"enable_client_credentials":"expected a boolean"}},"code":2}

### Steps To Reproduce

1. create an API 
http POST localhost:8001/apis name=test1 upstream_url=http://app.v1.service uris=/v1/test

2. enable OAuth plugin 
http post localhost:8001/apis/test1/plugins name=oauth2 config.enable_client_credentials=true

### Additional Details & Logs

- Kong version 0.15.0
- Kong db: postgres 9.5
- Additional logs:
[28/Jan/2019:14:50:56 +0000] "POST /apis/test1/plugins HTTP/1.1" 400 134 "-" "HTTPie/1.0.2"
kong_1 
[28/Jan/2019:14:59:05 +0000] "POST /apis/test1/plugins HTTP/1.1" 400 192 "-" "curl/7.54.0"

- Kong configuration: default  
- Operating system: docker 0.15.0-centos (docker-compose)

**Comment by p0pr0ck5** [comment]
Hi,

For the `http` utility, I've found you need to enable form-encoded request bodies instead of the default JSON body. Try:

`http post localhost:8001/apis/test1/plugins name=oauth2 config.enable_client_credentials=true -f`

**Comment by milaw** [comment]
Thanks for your comment.
I tried and it's not working:

http post localhost:8001/apis/test1/plugins name=oauth2 config.enable_client_credentials=true -f
http post localhost:8001/apis/test1/plugins name=oauth2 config.enable_client_credentials=true --form

HTTP/1.1 400 Bad Request
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 192
Content-Type: application/json; charset=utf-8
Date: Tue, 29 Jan 2019 06:59:52 GMT
Server: kong/0.15.0

{
    "code": 2,
    "fields": {
        "config": {
            "enable_client_credentials": "expected a boolean"
        }
    },
    "message": "schema violation (config.enable_client_credentials: expected a boolean)",
    "name": "schema violation"
}

Also, I wrote a microservice (typescript) in charge of provisioning the gateway according to specific logic. 
my code looks like this:


```
const params = {
            "name": "oauth2",
            "config.mandatory_scope": false,
            "config.enable_client_credentials": true,
            "config.hide_credentials": false,
            "config.token_expiration": 7200,
            "config.global_credentials": true,
  } ;
return await this.request("POST", `apis/${apiName}/plugins`, params) 

```
Works well with the previous versions.

**Comment by benbenqiang** [comment]
I had the same issue

**Comment by thibaultcha** [comment]
Fixed by #4272. We will ship a 0.15.1 release in the near future (we don't have a precise date yet), as we are working towards 1.1, we will continue pushing fixes for the 1.0.x cycle, and include all of the relevant fixes to 0.15.1. Feel free to apply the patch manually for the time being, if you can.

Thank you for the report!
